src/gui/dialog/toolbarCustomize/ToolbarAdapter.cpp

```diff
@@ -325,3 +299,18 @@ void ToolbarAdapter::toolbarDragDataReceivedCb(GtkToolbar* toolbar, GdkDragConte
         g_warning("toolbarDragDataReceivedCb: ToolItemType %i not handled!", d->type);
     }
 }
+
+auto ToolbarAdapter::toolbarGetDropIndex(GtkToolbar* toolbar, gint x, gint y, bool horizontal) -> gint {
+    // gtk_toolbar_get_drop_index does not use the correct coordinate system for the y coordinate
+    // since the y coordinate is only relevant in vertical toolbars the transformation is NOT required for horizontal
+    // toolbars
+    if (horizontal) {
+        return gtk_toolbar_get_drop_index(toolbar, x, y);
+    } else {
+        gint wx = 0;
+        gint wy = 0;
+        gtk_widget_translate_coordinates(GTK_WIDGET(toolbar), gtk_widget_get_toplevel(GTK_WIDGET(toolbar)), x, y, &wx,
+                                         &wy);
+        return gtk_toolbar_get_drop_index(toolbar, x, wy);
```

Comment:
```suggestion
        return gtk_toolbar_get_drop_index(toolbar, wx, wy);
```

I would feel more comfortable if you used both translated coordinates here. This might be a really hard to find bug, if at some point the x-coordinate becomes important (though I do not know if this would ever happen)